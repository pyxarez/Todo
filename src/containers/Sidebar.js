import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';

import './Sidebar.res/style.css';

export class Sidebar extends Component {
  static propTypes = {
    categoryContainer: PropTypes.func.isRequired,
    categories: PropTypes.arrayOf(PropTypes.object.isRequired).isRequired,
    currentCategoryId: PropTypes.number,
    currentTaskId: PropTypes.number,
  }

  _renderCategoryList = (CategoryContainer, categories = this.props.categories) => {
    return categories.map((category) => {
      const nestedCategories = category.nested;
      const id = parseInt(category.id, 10);

      return (
        <CategoryContainer
          key={id}
          id={id}
          title={category.title}
          {...this.props}>
          {category.nested.length > 0 &&
            this._renderCategoryList(CategoryContainer, nestedCategories)}
        </CategoryContainer>
      )
    });
  }

  render() {
    const {
      categoryContainer,
      categories
     } = this.props;
    return (
      <div className="my-sidebar-component">
        <div className="sidebar">
          {categories.length > 0 &&
            this._renderCategoryList(categoryContainer)}
        </div>
      </div>
    );
  }
}

const mapStateToProps = ({ categories }) => {
  return {
    categories
  }
}

export default connect(mapStateToProps)(Sidebar);
