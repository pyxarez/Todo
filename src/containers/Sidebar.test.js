import React, { Component } from 'react';
import { shallow } from 'enzyme';

import { Sidebar } from './Sidebar';

const setup = () => {
  const props = {
    categoryContainer: () => (<div className="test-category-container"></div>),
    categories: [
      {
        id: 0,
        title: 'The first title',
        nested: [
          {
            id: 1,
            title: 'The second title',
            nested: []
          }
        ]
      },
      {
        id: 2,
        title: 'Ther third title',
        nested: []
      }
    ],
    currentCategoryId: 1,
    currentTaskId: 1
  }
  const wrapper = shallow(<Sidebar {...props} />);

  return {
    props, 
    wrapper
  }
};

describe('Containers :: Sidebar', () => {
  it('should render self and 3 categories', () => {
    const { wrapper } = setup();
    
    expect(wrapper).toMatchSnapshot();
  });
  

  it('should render self', () => {
    const { props } = setup();
    const newProps = {
      ...props,
      categories: []
    };
    const wrapper = shallow(<Sidebar {...newProps}/>);

    expect(wrapper).toMatchSnapshot();
  });
});