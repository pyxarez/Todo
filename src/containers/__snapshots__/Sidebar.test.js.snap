exports[`Containers :: Sidebar should render self 1`] = `
<div
  className="my-sidebar-component">
  <div
    className="sidebar" />
</div>
`;

exports[`Containers :: Sidebar should render self and 3 categories 1`] = `
<div
  className="my-sidebar-component">
  <div
    className="sidebar">
    <categoryContainer
      categories={
        Array [
          Object {
            "id": 0,
            "nested": Array [
              Object {
                "id": 1,
                "nested": Array [],
                "title": "The second title",
              },
            ],
            "title": "The first title",
          },
          Object {
            "id": 2,
            "nested": Array [],
            "title": "Ther third title",
          },
        ]
      }
      categoryContainer={[Function]}
      currentCategoryId={1}
      currentTaskId={1}
      id={0}
      title="The first title">
      <categoryContainer
        categories={
          Array [
            Object {
              "id": 0,
              "nested": Array [
                Object {
                  "id": 1,
                  "nested": Array [],
                  "title": "The second title",
                },
              ],
              "title": "The first title",
            },
            Object {
              "id": 2,
              "nested": Array [],
              "title": "Ther third title",
            },
          ]
        }
        categoryContainer={[Function]}
        currentCategoryId={1}
        currentTaskId={1}
        id={1}
        title="The second title" />
    </categoryContainer>
    <categoryContainer
      categories={
        Array [
          Object {
            "id": 0,
            "nested": Array [
              Object {
                "id": 1,
                "nested": Array [],
                "title": "The second title",
              },
            ],
            "title": "The first title",
          },
          Object {
            "id": 2,
            "nested": Array [],
            "title": "Ther third title",
          },
        ]
      }
      categoryContainer={[Function]}
      currentCategoryId={1}
      currentTaskId={1}
      id={2}
      title="Ther third title" />
  </div>
</div>
`;
